core/src/main/java/org/infinispan/stream/impl/DistributedCacheStream.java

```diff
@@ -83,15 +82,16 @@
 /**
  * Implementation of {@link CacheStream} that provides support for lazily distributing stream methods to appropriate
  * nodes
+ * @param <Original> the original type of the underlying stream - normally CacheEntry or Object
```

Comment:
The type is exposed only for some type checks. To be honest the typing is not completely done here as the IntermediateOperation while having types on them are not provided (since when I add them I didn't have a good way to do it) I do now, but it requires more changes I didn't want to put in here yet.

https://github.com/infinispan/infinispan/pull/5794/files#diff-c37d0eb44c3ac6aad4102f7b81ec346dR278
https://github.com/infinispan/infinispan/pull/5794/files#diff-c37d0eb44c3ac6aad4102f7b81ec346dR129